walkeeper/src/timeline.rs

```diff
@@ -232,6 +232,7 @@ impl TimelineTools for Option<Arc<Timeline>> {
     /// Find last WAL record. If "precise" is false then just locate last partial segment
     fn find_end_of_wal(&self, data_dir: &Path, precise: bool) -> (Lsn, TimeLineID) {
         let seg_size = self.get().get_info().server.wal_seg_size as usize;
+        assert!(seg_size > 0);
```

Comment:
I think `wal_seg_size` comes from the compute node. We should treat that as untrusted data; a broken or malicious compute node could send a 0 seg size and cause the walkeeper to PANIC. Should have non-assert sanity checks somewhere.

Reply:
Situation with wal_seg_size is the following:walkeeper takes this value from compute node, pageserver takes default value (16Mb) specified in pg_constants.rs
From my point of view we should accept clients with different values f wal_seg_size (certainly it should be verified).
But once again it requires a logt of changes in pageserver code which I prefer to do in main branch and not in nonrel_wal branch.